### ISSUE TITLE
Running on node: "self is not defined"

### ISSUE THREAD

**Issue by patkar** [platformsh/platformsh-client-js#2]
Hey, I wanted to execute this client in node (on AWS Lambda, but has problems with local node too), but I get the following error:

```json
{
    "errorMessage": "self is not defined",
    "errorType": "ReferenceError",
    "stackTrace": [
        "t (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:411)",
        "Object.<anonymous> (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:7861)",
        "t (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:411)",
        "Object.e.__esModule.default (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:76957)",
        "t (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:411)",
        "Object.defineProperty.value (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:768)",
        "/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:778",
        "r.(anonymous function).i (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:143)",
        "Object.<anonymous> (/var/task/node_modules/platformsh-client/lib/platform-api.min.js.js:1:288)"
    ]
}
```

In this case my code is not even executed, it already fails out during the import/require statement (because they are sync).

The problem is the `isomorphic-fetch` package, and their `fetch-npm-browserify.js` get's compiled into the min.js when using webpack:

Seems like webpack is using the `browser` reference, instead of the `main` when building the package:

```json
{
  "name": "isomorphic-fetch",
  "version": "0.0.0",
  "description": "Isomorphic WHATWG Fetch API, for Node & Browserify",
  "browser": "fetch-npm-browserify.js",
  "main": "fetch-npm-node.js",
}
```

And they are blindly accessing the self object instead of checking if it's ever available... and self doesn't exist on node:

```js

// the whatwg-fetch polyfill installs the fetch() function
// on the global object (window or self)
//
// Return that as the export for use in Webpack, Browserify etc.
require('whatwg-fetch');
module.exports = self.fetch.bind(self);
```

**Comment by patkar** [comment]
Gotcha, the default target on webpack is `web`, thus it will use the `browser` property on a package if it exists.

So this package needs an extra node build, to be truly isomorphic :)

**Comment by ydeshayes** [comment]
Thanks a lot for the report and the MR! I will add some test for the nodejs side.

**Comment by ydeshayes** [comment]
Published in 0.1.23 version :)
